docs/user_guide/fairness_in_machine_learning.rst

```diff
@@ -141,6 +141,187 @@ which groups these values were observed, as well as the difference and ratio
+in underlying social values and incentives of a given social system; for
+instance, by increasing perceived value of quantifiable metrics over
+non-quantifiable ones.
+
+Example: consider the problem of banks deciding whether an individual should
+be approved for a loan. Prior to using machine learning algorithms
+to compute a "score", banks might rely on loan officers that engage in
+conversations with clients, recommend a plan based on their unique
+situation, and discuss with other team members to obtain feedback.
+By introducing an algorithm, it is possible that loan officers may limit
+their conversations with team members and clients, assuming the algorithm's
+recommendations are good enough without those additional sources of information.
+
+To avoid this pitfall, we must be aware that once a technology is incorporated
+into a social context, new groups may reinterpret it differently. We should
+adopt "what if" scenarios to envision how the social context might change
+after introducing a model, including how it may change the power dynamics of
+existing groups in that context, or how actors might change their behaviors to
+game the model.
+
+.. _formalism_trap:
+The Formalism Trap
+^^^^^^^^^^^^^^^^^^
+
+Many tasks of a data scientist involve some form of formalization: from
+measuring real-world phenomena as data to translating business KPI's
+and constraints into metrics, loss functions, or parameters. We fall into the
+formalism trap when we fail to account for the full meaning of social
+concepts like fairness.
+
+Fairness is a complex construct that is contested: different people may
+have different ideas of what is fair in a particular scenario. While
+mathematical fairness metrics may capture some aspects of fairness, they
+fail to capture all relevant aspects. For example, group fairness metrics
+do not account for differences in individual experiences, nor do they
+account for procedural justice.
+
+In some scenarios, fairness metrics such as demographic parity and equalized
+odds cannot be satisfied at the same time. At a first glance, this may appear
+to be a mathematical problem. However, the conflict is actually grounded in
+different understandings of what fairness is. Consequently, there is no
+mathematical approach to solve the conflict. Instead we need to decide which
+metrics might be appropriate for the situation at hand, keeping in mind the
+limitations of a mathematical formalization. In some cases, there may be no
+suitable metric.
+
+Some reasons why we fall into this trap are because fairness is
+context-dependent, because it is open to contestation by different groups
+of people, and because there are differences between ways of thinking about
+fairness between the legal world (i.e., fairness as procedural) and the fair-ML
+community (i.e., fairness as outcome-based).
+
+Where mathematical abstraction encounters a limitation is when
+capturing information regarding procedurality (for example, how do judges
```

Comment:
Ah, I think I had left a comment previously about the sequencing between this paragraph and the previous one. 

The previous one is 123 (contextual; contested; procedural) and this one is 312 (procedural; contextual; contested)...

Can we make both of them the same sequence? That'll make it easier for readers to follow

Reply:
Reordered